compiler/rustc_span/src/edit_distance.rs

```diff
@@ -198,12 +199,27 @@ fn find_best_match_for_name_impl(
         } {
             Some(0) => return Some(*c),
             Some(d) => {
-                dist = d - 1;
-                best = Some(*c);
+                if use_substring_score {
+                    dist = d;
+                    next_candidates.push(*c);
+                    best = Some(*c);
+                } else {
+                    dist = d - 1;
+                    best = Some(*c);
+                }
```

Comment:
Could you explain why we introduce a different behaviour here?
In particular in the assignment to `dist`?

Reply:
updated it!